### DISCUSSION TITLE
All my NextJs API suddenly returns 504 timeout

### DISCUSSION THREAD

**Discussion by robertwt7** [vercel/community#2447] [Help] [answered]
### Summary

Starting from may 12th, all my nextJs API suddenly returns 504 timeout.

![image](https://github.com/vercel/community/assets/15647967/e660fe42-9fbb-449a-a388-d25e68804827)

i haven't had any changes to the API during these times.

I have also checked that the api exist in the source code:
![image](https://github.com/vercel/community/assets/15647967/e0ae0a3a-b2f5-4e38-ac7f-ecdd57dcdcf4)

Also the Log doesn't tell me much as well:
![errorVercel](https://github.com/vercel/community/assets/15647967/272fdbf2-fbc6-47f6-846d-05284532726f)

I have tried to build the app with a lot of `console.log` but it doesn't return anything..

i am also using turbo build on my project. the url is https://app.hublr.co

does anyone have any idea what's going on here? i have been scratching my head the whole day now trying to solve this issue.
everything works perfectly fine locally, my backend is also working, my db in supabase is also online..

**Comment by amyegan** [comment]
Sorry you're running into timeout errors, @robertwt7. It looks like this is related to auth, and possibly NextAuth, based on the error log you shared. It could be something as simple as an expired token for one of your auth services, so the first thing to check is that all of the necessary environment variables are present and valid. 

If that doesn't solve it, you should be able to get a bit more info by clicking on an individual log row to expand the details.

**Comment by robertwt7** [comment]
Thankyou for replying @amyegan , i don't think this is related to auth. the `/api/advisors` is only an api call to airtable, it works locally as well but this endpoint is also returning 504 for me.

I have checked if all the env variables are valid. also the individual log only says `info` or `undefined`

let me try to copy more of the logs

**Comment by robertwt7** [comment]
is there any known issue that I should be aware off? @amyegan 

everything works locally and it still doesn't work on vercel

steps i have tried:
- delete project and redeploy
- update all env variables
- tested api locally, it works both for my API that shoots to my backend and API that gets data from airtable.

i'm not sure what else to do next as the logs just says `undefined`

**Reply by amyegan** [reply]
If the project is on a Hobby account, the [Serverless Function Execution Timeout](https://vercel.com/docs/concepts/limits/overview#serverless-function-execution-timeout) limit is 10 seconds. If the function takes longer than that to respond, you'll see the timeout error. Since it works in your local environment which doesn't have the timeout limit, I think the problem is that the serverless function is taking too long to execute. The logs should reveal some info about how long the request took.

**Reply by jswift24** [reply]
I have a similar problem. Our app calls the ChatGPT API. Sometimes it takes 20-30 seconds to come back. When that happens locally, our app works fine. When it happens on Vercel's hosted site, our app gives a 504 (timeout) error. How can I change the timeout to 60 seconds on Vercel? I've already upgraded to the pro plan. Thanks!

**Reply by robertwt7** [reply]
the problem is not supposed to be the timeout. i tried connecting my local front end to my backend on the server and it takes 9ms to do all the api request. I am simply plugging my local front end to my remote backend, with the same code.

it should never take more than 10 seconds for my API as it is only querying from airtable, and the other one is to login

![image](https://github.com/vercel/community/assets/15647967/88a4ff5a-f7e6-43f3-8bea-0c2a08ddbcf0)

**Reply by robertwt7** [reply]
it was some turbo error

the default in vercel is:
`cd ../.. && turbo run build --filter={/apps/web}`

it should be `cd ../.. && turbo run build` because i have some dependencies on other build apparently :/

**Reply by robertwt7** [reply]
omg it doesn't work again after i moved the whole project to my org, with the exact same configuration. i really don't know what's wrong with the serverless functions

**Reply by robertwt7** [reply, marked as answer]
FINALLY I FIND THE ANSWER

its related to mui icons omgg. not sure why it has something to do with the API 

https://github.com/netlify/next-runtime/issues/1174#issuecomment-1087630605

**Reply by amyegan** [reply]
I'm glad you found a solution! This problem had me stumped, and I never would have figured out that it was caused by and icon library based on the screenshots. 😅 

Seems like the problem may be related to bundle size. The [MUI Icons documentation](https://mui.com/material-ui/icons/#usage) notes that `import AccessAlarmIcon from '@mui/icons-material/AccessAlarm'` is safer for bundle size than `import { AccessAlarm } from '@mui/icons-material'`. In any case, I'm just happy to see you got it working!
